### PR TITLE
Data.ByteString.Lazy.Builder -> Data.ByteString.Builder

### PR DESCRIPTION
**Pull request by bgamari**
The former was deprecated in 0.10.10.

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog (add file to `changelog.d` directory).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by phadej** [comment]
Why introduce CPP, is there some `-Werror` annoyance somewhere?

**Comment by bgamari** [comment]
Because the module is not present in earlier versions of `bytestring`.

**Comment by phadej** [comment]
Yes, I'm wondering what is the problem with using old name for as long as it exists?

**Comment by bgamari** [comment]
> Yes, I'm wondering what is the problem with using old name for as long as it exists?

The module is marked as deprecated and consequently `Cabal` is not `-Werror` clean with `bytestring-0.10.12.0`.

**Comment by phadej** [comment]
Fun fact. PVP says (atm)

> 7. Deprecation. Deprecated entities (via a DEPRECATED pragma) SHOULD be counted as removed for the purposes of upgrading the API, because packages that use -Werror will be broken by the deprecation. In other words the new A.B SHOULD be greater than the previous A.B.

TL;DR, don't deprecate, just remove straight away.

**Comment by phadej** [comment]
Edited and merged in https://github.com/haskell/cabal/pull/7049
